encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/RsaAlgorithm.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.config.crypto.algorithm;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.palantir.config.crypto.EncryptedValue;
+import com.palantir.config.crypto.KeyPair;
+import com.palantir.config.crypto.KeyWithAlgorithm;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyFactory;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+
+public final class RsaAlgorithm implements Algorithm {
+
+    public static final String ALGORITHM_TYPE = "RSA";
+    private static final Charset charset = StandardCharsets.UTF_8;
+
+    private Cipher getUninitializedCipher() throws NoSuchAlgorithmException,
+        NoSuchPaddingException, NoSuchProviderException {
+        return Cipher.getInstance(ALGORITHM_TYPE);
+    }
+
+    private int getKeyLength() throws NoSuchAlgorithmException {
+        return Math.min(Cipher.getMaxAllowedKeyLength(ALGORITHM_TYPE), 512);
```

Comment:
hm, I think we want to allow RSA keys to exceed 512 bits because a string encoded by an RSA key must be shorter than the key length, and we might need to encode long strings, and some users might choose larger keys.